README.md

```diff
@@ -236,6 +236,68 @@ In order to improve the performance of the first connection per request, we use
 By default, laravel-spanner uses [Filesystem Cache Adapter](https://symfony.com/doc/current/components/cache/adapters/filesystem_adapter.html) as the caching pool. If you want to use your own caching pool, you can extend ServiceProvider and inject it into the constructor of `Colopl\Spanner\Connection`.
 
 
+
+### 'Session not found' exception handling
+There are a few cases when a 'Session not found' error can
+[happen](https://cloud.google.com/spanner/docs/sessions#handle_deleted_sessions):
+
+ - Scripts that idle too long - for example, a [Laravel queue worker](https://laravel.com/docs/9.x/queues#the-queue-work-command)
+or anything that doesn't call Spanner frequently enough (more than once an hour).
+ - The session is more than 28 days old.
+ - Some random flukes on Google's side.
+
+The errors can be handled by one of the supported modes:
+
+- **MAINTAIN_SESSION_POOL** - When the 'session not found' error is encountered, the library tries to disconnect,
```

Comment:
I think the behavior for `CLEAR_SESSION_POOL` is good enough and this mode unnecessary.
By only having one way of doing things would make the whole thing simpler and easier to understand.
What do you think?

Reply:
Clearing session pool (while really is needed in very rare occasions)  is very 'hard' operation, which implicitly affects all processes that use session poll, i.e. for example if there 50 (httpd/php-fpm) workers that share session pool, then after clearing they all simultaneously will receive 'session not found' (because clearing deletes sessions) and end with restarting potentially heavy and long running scripts
so MAINTAIN_SESSION_POOL is the most safe mode - it still handles most of the cases (like idle workers), but do not introduce hard reset of sessions

so there are possible options
- left MAINTAIN_SESSION_POOL as safest way for users who doesn't want running into potentially risking hard reset
- left it and make it default 
- remove it, but provide users some level of control 
```
function setClearSessionPoolOnSessionNotFoundError(bool $value)
```
though it seems ugly and less clean

Reply:
I understood your concern. I now believe this can be as is.

However, now I'm thinking the opposite. Is `CLEAR_SESSION_POOL` necessary?

Wouldn't each session be retried eventually anyway?

Reply:
> Is CLEAR_SESSION_POOL necessary
yes, it is necessary at least in my use case - without it there is error once a week or two and maintain() didn't help

Reply:
So I'm guessing the error still occurred because in a case where `MAINTAIN_SESSION_POOL` is used, the handler only handles the error once, and upon reconnection, the logic probably choose another session that was also expired and errored out.
Is my understanding correct?

Reply:
no, There are rare cases when google spanner deletes a session entirely, while the session is still accounted in session pool as active, so maintain and reconnection just reload it again and again. and the only way to fix it is deletion of the cache   file manually (or via code, or restarting docker/pod)

Reply:
`MAINTAIN_SESSION_POOL` set as default mode